airflow/providers/microsoft/psrp/hooks/psrp.py

```diff
@@ -31,13 +33,19 @@ class PSRPHook(BaseHook):
     Hook for PowerShell Remoting Protocol execution.
 
     The hook must be used as a context manager.
+
+    :param psrp_conn_id: Required. The name of the PSRP connection.
+    :type psrp_conn_id: str
+    :param logging: If true (default), log command output and streams during execution.
+    :type logging: bool
```

Comment:
I wonder if we should instead make `poll_interval` configurable. `logging` is not a great name IMO since it does not show the implication of the process being polled, which I believe can be a performance impact. Allowing the user to configure `poll_interval` can make this more visible, and `poll_interval=None` (or `0` or `-1`?) would be a more natural value to disable polling altoghether.

BTW, is logging still possible without polling? i.e. instead of outputting eagerly, gather all outputs and log them after `end_invoke` instead.

Reply:
The logs are always available at the end – I don't change any of the recorded information but just keep a pointer to how far into the various streams I have forwarded to Python's logging system.

Reply:
@uranusjr I looked into this in more detail now and the way it works is that there is an _operational timeout_ after which polling must return with an updated status (or a failure) – but it can return earlier.

I did a test where I issued the PowerShell cmdlet `Start-Sleep -s 3; Write-Host "Hello"` (sleep for three seconds) in a loop. Even though the default operational timeout is 20 seconds, each polling action returned with a new status after 3 seconds.

I have updated the code to reflect this new understanding and instead of a polling interval there is now an overrideable operational timeout. For example, setting it to 1 in the above example would mean that every second we get a status update, but only every 3 seconds do we get a "Hello" information stream record.